src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs

```diff
@@ -208,7 +210,20 @@ public CookieContainer CookieContainer
 
 		public ClientCertificateOption ClientCertificateOptions { get; set; }
 
-		public X509CertificateCollection? ClientCertificates { get; set; }
+		private X509CertificateCollection? _clientCertificates;
+		public X509CertificateCollection? ClientCertificates
+		{
+			get
+			{
+				if (ClientCertificateOptions != ClientCertificateOption.Manual) {
+					throw new InvalidOperationException ($"Enable manual options first on {nameof (ClientCertificateOptions)}");
```

Comment:
I believe that this is somewhat *contrary* to the [Framework design guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/); from [Property Design](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/property):

> ✔️ DO allow properties to be set in any order even if this results in a temporary invalid state of the object.
> 
> It is common for two or more properties to be interrelated to a point where some values of one property might be invalid given the values of other properties on the same object. In such cases, exceptions resulting from the invalid state should be postponed until the interrelated properties are actually used together by the object.

I think it would thus be preferable to *delay* the "consistency check" until…

Reply:
First of all, thanks for the detailed feedback, @jonpryor!

You have a great point here, this design choice isn't great. I'm following [`HttpClientHandler`](https://github.com/dotnet/runtime/blob/e8b89a3fde2911c6cbac0488bf82c74329a7224a/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L282-L294) which also throws.

Reply:
On the one hand, following what `HttpClientHandler` is doing makes sense.

On the other hand, [I'm not convinced that actually fixes the implicit ordering issues](https://github.com/xamarin/xamarin-android/pull/8961/files#r1637303201).

A quick glance through the [`HttpClientHandler` unit tests for `ClientCertificates`](https://github.com/dotnet/runtime/blob/a162a1fd37d5b205267d4aa510c1bb31e2d8435c/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.ClientCertificates.cs) suggests that this ordering issue is not tested at all.  *Fortunately*, I don't see any issues on dotnet/runtime about this either…

Reply:
The required order makes sense to me. The error you get is helpful too and expected